internal/plugin/connectors/tcp/mssql/mock/options.go

```diff
@@ -26,76 +26,43 @@ type MSSQLConnectorCtorOptions struct {
 	ServerLoginResponse *mssql.LoginResponse
 }
 
-// MSSQLConnectorCtorOption is the 'functional option' complement to
-// MSSQLConnectorCtorOptions.
-type MSSQLConnectorCtorOption func(*MSSQLConnectorCtorOptions)
-
-// MSSQLConnectorCtor is a 'functional option' of types.ConnectorOption. It generates
-// a customizable MSSQLConnectorCtor using MSSQLConnectorCtorOptions, then sets it on the
-// ConnectorOption passed to it.
-//
-// Empty values of MSSQLConnectorCtorOption result default:
-// ServerPreloginResponse => map[uint8][]byte{}
-// ServerLoginResponse => &mssql.LoginResponse{}
-// BackendConn => nil
-// BackendConn => nil
-// ClientLoginRequestPtr => nil (this means ClientLoginRequest is read from the channel to
-// a vacuum)
-func MSSQLConnectorCtor(setters ...MSSQLConnectorCtorOption) types.ConnectorOption {
-	args := &MSSQLConnectorCtorOptions{}
-
-	for _, setter := range setters {
-		setter(args)
-	}
+// NewMSSQLConnectorCtor is the default constructor for MSSQLConnectorCtor
+var NewMSSQLConnectorCtor = MSSQLConnectorCtor{
```

Comment:
I'm confused about this. I don't see it used anywhere ?

Reply:
Essentially this is a constructor used everywhere you made the comments above - it fills in default values so we don’t have to worry about checking for nil. That was part of Jonah’s comment.